server/jetstream_api.go

```diff
@@ -118,6 +118,12 @@ const (
 	JSApiMsgGet  = "$JS.API.STREAM.MSG.GET.*"
 	JSApiMsgGetT = "$JS.API.STREAM.MSG.GET.%s"
 
+	// JSDirectMsgGet is the template for non-api layer direct requests for a message by its stream sequence number or last by subject.
+	// Will return the message similar to how a consumer receives the message, no JSON processing.
+	// If the message can not be found we will use a status header of 404. If the stream does not exist the client will get a no-responders or timeout.
+	JSDirectMsgGet  = "$JS.DS.GET.*"
+	JSDirectMsgGetT = "$JS.DS.GET.%s"
```

Comment:
doesnt domains allow 2 domains to host the same name stream? But the core subject space is shared by all domains in an account? So just saying stream name could lead to issues right?

Not that up to speed with domains so might just be confused

Reply:
Fair question.

Reply:
This needs to be thought through a bit more, but even now if those streams have the same subjects they will be placed into both.

I also think we should place the direct get in a DQ group by default such that replicas can share the load.

This is experimental for now as we settle in the API and subjects and mechanics. Will merge.

Reply:
Hmm the domain scenario where 2 streams have the same name and subject space - a monumentally bad idea - isn’t about replicas it’s about like regional to hub migration/consolidation. 

So it’s not about replicas sharing the load. Data might just not be there. 

Wirh this I would say direct must not be default behaviour but sometning you opt in when you really grok your setup?

Reply:
Direct is opt-in, agree you don't want to cross over with Domains, just saying that issues already is present with ingest subjects.